lib/p5.play.js

```diff
@@ -4189,25 +4199,35 @@ function Animation(pInst) {
       pInst.push();
       pInst.imageMode(CENTER);
 
-      pInst.translate(this.xpos, this.ypos);
+      var xTranslate = this.xpos;
+      var yTranslate = this.ypos;
+      var image = this.images[frame];
+      var frame_info = this.spriteSheet && image;
+
+      // Adjust translation if we're dealing with a texture packed spritesheet
+      // (with sourceW, sourceH, sourceX, sourceY props on our images array)
+      if (frame_info) {
+        var missingX = (frame_info.sourceW || frame_info.width) - frame_info.width;
+        var missingY = (frame_info.sourceH || frame_info.height) - frame_info.height;
+        xTranslate -= (missingX - (frame_info.sourceX || 0)) / 2;
+        yTranslate -= (missingY - (frame_info.sourceY || 0)) / 2;
```

Comment:
Computing the "missing" padding here and dividing by two is a red flag for me. I think there's an assumption here that the way TexturePacker trims sprites is always symmetrical, and the center of the trimmed frame is an appropriate anchor.  I don't think this is true - the second frame of sample JSON you provided shows a trimmed frame that is _not_ centered in the source frame.
```json
{
	"filename": "Unicorn_Floss_01.png",
	"frame": {"x":613,"y":269,"w":120,"h":269},
	"rotated": false,
	"trimmed": true,
	"spriteSourceSize": {"x":92,"y":24,"w":120,"h":269},
	"sourceSize": {"w":300,"h":300}
}
```

This frame has 92px trimmed on the left, 88px trimmed on the right, 24px trimmed on the top and 7px trimmed on the bottom.  For some reason it looks like you're 'restoring' 44px and 3.5px of trimmed space.

```
frame_info.width = 120
frame_info.sourceX = 92
frame_info.sourceW = 300
missingX = 180
xTranslate -= 44

frame_info.height = 269
frame_info.sourceY = 24
frame_info.sourceSize = 300
missingY = 31
yTranslate -= 3.5
```

Ideally the solution here requires less computation; the `frame` given by TexturePacker should be all the information we need for the blit from the source image, and the `spriteSourceSize` x and y should be the unscaled destination offset from the sprite's anchor point (untrimmed top-left).

I haven't quite loaded this all into my brain to write the correct approach here.  Let me get into the office and take another look today.

Reply:
@islemaster : In looking at this more carefully, this step is essentially accounting for missing pixels on the right or at the bottom. For the frame in question, we are missing 88 pixels on the right and 7 pixels on the bottom. By translating by -44, -3.5, we can draw as if there are no missing pixels on the right or bottom. However, the missing pixels on the left and the top are dealt with below (currently line 4226). It's probably better to do both in the same place to ensure that everything works properly with respect to rotation. I think I'll move the code from line 4226 up into the `xTranslate`, `yTranslate` computation, because that should better preserve the rotation behavior.